packages/client/src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,17 @@
+import { Role } from '@fullstack/sdk';
+import { createContext, useMemo, useState } from 'react';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const AuthContext = createContext<{ auth: { isLoggedIn?: boolean; userRole?: Role }; setAuth: any }>({
+  auth: {},
+  setAuth: null,
+});
+
+const AuthProvider = ({ children }: { children: JSX.Element[] | JSX.Element }) => {
+  const [auth, setAuth] = useState({});
```

Comment:
Po przejrzeniu wszystkich `setAuth` w codebase'ie:
```TS
setAuth({ isLoggedIn: false, userRole: undefined });
setAuth({ isLoggedIn: true, userRole });
// + początkowy stan:
auth: {},
```
Zdaje się, że `isLoggedIn` jest bezpośrednio zależne od `userRole`:
```TS
const [userRole, setUserRole] = useState<Role | undefined>(undefined);
const isLoggedIn = userRole !== undefined;
```
(wolę `null` bo jest jasne rozgraniczenie między czymś co nie jest zdefiniowane, a czymś co jest zdefiniowane jako puste, ale to na marginesie bo działa w obu wersjach)

Wtedy możesz stan całego kontekstu trzymać w ww. `userRole`, jako `value` mieć obiekt: `{ userRole, setUserRole, isLoggedIn: userRole !== null }`.

```TS
// To:
setAuth({ isLoggedIn: false, userRole: undefined });
// Zmienić w to:
setUserRole(null);
// To:
setAuth({ isLoggedIn: true, userRole });
// Zmienić w to:
setUserRole(userRole);
```

Cały pomysł w jednym miejscu (wliczając zmiany typizacji itd.):
```TS
const AuthContext = createContext({
  userRole: null as Role | null,
  setUserRole: (() => {}) as (role: Role | null) => void,
  isLoggedIn: false,
});

const AuthProvider = ({ children }: { children: ReactNode }) => {
  const [userRole, setUserRole] = useState<Role | null>(null);
  const value = useMemo(() => ({ userRole, setUserRole, isLoggedIn: userRole !== null }), [userRole, setUserRole]);
  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
};

const useAuth = () => useContext(AuthContext);

export { AuthProvider, useAuth };
```